src/openms/source/ANALYSIS/QUANTITATION/AbsoluteQuantitation.cpp

```diff
@@ -641,5 +641,58 @@ namespace OpenMS
     }
   } 
 
-} // namespace
+  void AbsoluteQuantitation::getOptimizedCalibrationCurve(
+    const String& component_name,
```

Comment:
1. I would recommend using the inputs (i.e., component_name, and component_concentrations) to make a dictionary, which will then be the input to optimizeCalibrationCurve
2. I would recommend then taking the updated dictionary, and using that to update the input (i.e., component_concentrations)
This should prevent a lot of code re-use.

Reply:
really good idea